### DISCUSSION TITLE
How to make ErrorBoundary/useRouteError rerender?

### DISCUSSION THREAD

**Discussion by quangIle** [remix-run/remix#9593] [Q&A]
Hello, I have a case where I have a route's `loader` throw an error and the `ErrorBoundary` is rendered. But after that, if that `loader` is triggered again and return a successful response, the `ErrorBoundary` will not disappear (so that my route could be rendered instead). 
I know that trigger the whole web reload works but are there any different alternatives?

**Comment by kiliman** [comment]
As long as the loader doesn't return another error there shouldn't be a reason that the error boundary continues to render. 

Do you have a repro showing the problem?

**Reply by quangIle** [reply]
Thank you for your respond. After some investigate, I think my problem is that `loader`'s revalidation after `fetcher.Form` or `fetcher.submit` will not make error boundary re-render for an successful response.

Here is my reproduce: [example](https://stackblitz.com/edit/remix-run-error-boundary-with-fetcher?file=app%2Froutes%2Ferror._index.tsx)
Click the submit button will make a POST request to layout's action, then nested route's loader will run again but error boundary does not disappear.
